test/Microsoft.Sbom.Extensions.DependencyInjection.Tests/RemapComponentDetectionErrorsToWarningsLoggerTests.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Serilog;
+using Serilog.Events;
+using Serilog.Parsing;
+
+namespace Microsoft.Sbom.Extensions.DependencyInjection.Tests;
+
+[TestClass]
+public class RemapComponentDetectionErrorsToWarningsLoggerTests
+{
+    private const string TestPropertyKey = "TestPropertyKey";
+
+    private Mock<ILogger> loggerMock;
+    private Func<string> stackTraceProvider;
+    private int testStackTraceCount;
+    private string? testStackTraceReturnValue;
+
+    // We need a concrete implementation of MessageTemplateToken
+    internal class TestMessageTemplateToken : MessageTemplateToken
+    {
+        public override int Length => throw new NotImplementedException();
+
+        public override void Render(IReadOnlyDictionary<string, LogEventPropertyValue> properties, TextWriter output, IFormatProvider formatProvider = null) => throw new NotImplementedException();
+    }
+
+    // We need a concrete implementation of LogEventPropertyValue
+    internal class TestLogEventPropertyValue : LogEventPropertyValue
+    {
+        public override void Render(TextWriter output, string format = null, IFormatProvider formatProvider = null) => throw new NotImplementedException();
+    }
+
+    [TestInitialize]
+    public void InitializeDefaults()
+    {
+        loggerMock = new Mock<ILogger>(MockBehavior.Strict);
+        stackTraceProvider = TestGetStackTrace;
+        testStackTraceCount = 0;
+        testStackTraceReturnValue = null;
+    }
+
+    [TestMethod]
+    public void Constructor_LoggerIsNull_ThrowsNullArgumentException()
+    {
+        Assert.ThrowsException<ArgumentNullException>(() => new RemapComponentDetectionErrorsToWarningsLogger(null, stackTraceProvider));
+    }
+
+    [TestMethod]
+    public void Constructor_StackTraceProviderIsNull_ThrowsNullArgumentException()
+    {
+        Assert.ThrowsException<ArgumentNullException>(() => new RemapComponentDetectionErrorsToWarningsLogger(loggerMock.Object, null));
+    }
+
+    [TestMethod]
+    public void Write_LogEventLevelIsNotError_DoesNotCallStackTraceProvider_LogsEventAsSpecified()
+    {
+        var nonErrorLevels = Enum.GetValues(typeof(LogEventLevel)).Cast<LogEventLevel>().Where((x) => x != LogEventLevel.Error);
+        foreach (var level in nonErrorLevels)
+        {
+            var logEvent = GetLogEvent(level);
+            loggerMock.Setup(x => x.Write(logEvent)).Verifiable();
+            var logger = new RemapComponentDetectionErrorsToWarningsLogger(loggerMock.Object, stackTraceProvider);
+
+            logger.Write(logEvent);
+
+            loggerMock.VerifyAll();
```

Comment:
nit: our usual pattern is to have a test clean up method that verifies all of the mocks, so we don't have to remember to call verifyAll in every test separately. 
I also don't think the loggerMock.Reset is necessary, as the loggerMock is recreated before every test is run, right?

Reply:
Updated. In this case, since `loggerMock` is being recycled in a loop, the `VerifyAll` and `Reset` calls are needed. I suppose we could skip them on the last iteration, but that would just add code clutter without changing what is being tested. I added a comment about why the lines are present